scripts/calc_elf_addresses.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+
+import argparse
+from elftools.elf.elffile import ELFFile
+
+args = None
+def parse_args():
+	global args
+
+	parser = argparse.ArgumentParser(description='Helper utility to generate a linker command'
+                                         ' line with calculated ELF section addresses')
+
+	parser.add_argument("-f", "--file", required=True, type=str,
+						help='Object file name')
+	parser.add_argument("-t", "--text-addr", required=True, type=str,
+						help='.text section address')
+
+	args = parser.parse_args()
+
+def main():
+	parse_args()
+
+	elf = ELFFile(open(args.file, 'rb'))
+
+	text_addr = int(args.text_addr.replace("0x", ""), 16)
+
+	text_offset = elf.get_section_by_name('.text').header.sh_offset
+	rodata_offset = elf.get_section_by_name('.rodata').header.sh_offset
+	data_offset = elf.get_section_by_name('.data').header.sh_offset
+
+	upper = rodata_offset - text_offset + int(args.text_addr, 16) + 0xfff
+	rodata_addr = upper - (upper % 0x1000)
+
+	upper = data_offset - rodata_offset + rodata_addr + 0xf
+	data_addr = upper - (upper % 0x10)
+
+	print(f'-Wl,-Ttext=0x{text_addr:x} '
+              f'-Wl,--section-start=.rodata=0x{rodata_addr:x} '
+              f'-Wl,-Tdata=0x{data_addr:x}')
```

Comment:
going through a print statement like this is something that I believe is error prone and hard to debug.
What if warnings or other info is printed during python invocation ?